components/suggest/src/yelp.rs

```diff
@@ -0,0 +1,350 @@
+        scope.err_if_interrupted()?;
+        conn.execute(
+            "INSERT INTO yelp_modifiers(record_id, type, keyword) VALUES(:record_id, :type, :keyword)",
+            named_params! {
+                ":record_id": record_id.as_str(),
+                ":type": Modifier::PostModifier,
+                ":keyword": keyword,
+            },
+        )?;
+    }
+
+    for sign in &suggestion.location_signs {
+        scope.err_if_interrupted()?;
+        conn.execute(
+            "INSERT INTO yelp_location_signs(record_id, keyword, need_location) VALUES(:record_id, :keyword, :need_location)",
+            named_params! {
+                ":record_id": record_id.as_str(),
+                ":keyword": sign.keyword,
+                ":need_location": sign.need_location,
+            },
+        )?;
+    }
+
+    Ok(())
+}
+
+/// Fetch Yelp suggestion from given user's query.
+pub(crate) fn fetch_suggestion(
+    conn: &Connection,
+    query: &SuggestionQuery,
+) -> Result<Option<Suggestion>> {
+    if !query.providers.contains(&SuggestionProvider::Yelp) {
+        return Ok(None);
+    }
+
+    if query.keyword.len() > MAX_QUERY_LENGTH {
+        return Ok(None);
+    }
+
+    if !query.keyword.contains(' ') {
+        if !is_subject(conn, &query.keyword)? {
+            return Ok(None);
+        }
+
+        let builder = SuggestionBuilder {
+            query,
+            subject: &query.keyword,
+            pre_modifier: None,
+            post_modifier: None,
+            location_sign: None,
+            location: None,
+            need_location: false,
+        };
+        return Ok(Some(builder.into()));
+    }
+
+    // Find the location sign and the location.
+    let (query_without_location, location_sign, location, need_location) =
+        find_location(conn, &query.keyword)?;
+
+    if let (Some(_), false) = (&location, need_location) {
+        // The location sign does not need the specific location, but user is setting something.
+        return Ok(None);
+    }
+
+    if query_without_location.is_empty() {
+        // No remained query.
+        return Ok(None);
+    }
+
+    // Find the modifiers.
+    let (subject_candidate, pre_modifier, post_modifier) =
+        find_modifiers(conn, &query_without_location)?;
+
+    if !is_subject(conn, &subject_candidate)? {
+        return Ok(None);
+    }
+
+    let builder = SuggestionBuilder {
+        query,
+        subject: &subject_candidate,
+        pre_modifier,
+        post_modifier,
+        location_sign,
+        location,
+        need_location,
+    };
+    Ok(Some(builder.into()))
+}
+
+fn find_location(
+    conn: &Connection,
+    query: &str,
+) -> Result<(String, Option<String>, Option<String>, bool)> {
+    let query_with_spaces = format!(" {} ", query.trim());
+    let results: Vec<(usize, usize, i8)> = conn.query_rows_and_then(
+        "
+        SELECT
+          INSTR(:query, ' ' || keyword || ' ') AS sign_index,
+          LENGTH(keyword) AS sign_length,
```

Comment:
Ah, it might be the same answer of upper/lower cases. 
https://github.com/mozilla/application-services/pull/5951/files#r1445554770
Especially, we set `location_sign` that has `need_location` = 0 such as the `near by` to the `find_desc` parameter. So, to keep the string cases, I am doing it so. If we can use lowercase for the all characters for URL, yes, your suggestion looks much better!